src/core.jl

```diff
@@ -83,6 +100,30 @@ function apply(model::AbstractExplicitLayer, x, ps, st::NamedTuple)
     return model(x, ps, st)
 end
 
+function apply(model::AbstractExplicitLayer{true, false}, x, ps)
+    return apply(model, x, ps, NamedTuple())
+end
+
+function apply(model::AbstractExplicitLayer{false, true}, x, st::NamedTuple)
+    return apply(model, x, NamedTuple(), st)
+end
+
+function apply(model::AbstractExplicitLayer{false, false}, x)
+    return apply(model, x, NamedTuple(), NamedTuple())
```

Comment:
I think there is a disconnect between these functions and the original goal. The idea of having these properties in the type, would be to simplify user code. So a stateless layer can define `(::Layer)(x, ps)`. With this the user is still forced to define the original 3 argument api.